test_conformance/compiler/test_build_helpers.cpp

```diff
@@ -346,12 +351,19 @@ int test_get_program_info(cl_device_id deviceID, cl_context context, cl_command_
     error = clGetProgramInfo( program, CL_PROGRAM_DEVICES, sizeof( device1 ), &device1, NULL );
     test_error( error, "Unable to get device of program" );
 
-  /* Since the device IDs are opaque types we check the CL_DEVICE_VENDOR_ID which is unique for identical hardware. */
-  cl_uint device1_vid, deviceID_vid;
-  error = clGetDeviceInfo(device1, CL_DEVICE_VENDOR_ID, sizeof(device1_vid), &device1_vid, NULL );
-  test_error( error, "Unable to get device CL_DEVICE_VENDOR_ID" );
-  error = clGetDeviceInfo(deviceID, CL_DEVICE_VENDOR_ID, sizeof(deviceID_vid), &deviceID_vid, NULL );
-  test_error( error, "Unable to get device CL_DEVICE_VENDOR_ID" );
+    /* Object comparability test. */
+    test_assert_error(device1 == deviceID,
+                      "Unexpected result returned by CL_PROGRAM_DEVICES query");
+
+    /* Since the device IDs are opaque types we check the CL_DEVICE_VENDOR_ID
+     * which is unique for identical hardware. */
+    cl_uint device1_vid, deviceID_vid;
+    error = clGetDeviceInfo(device1, CL_DEVICE_VENDOR_ID, sizeof(device1_vid),
+                            &device1_vid, NULL);
+    test_error(error, "Unable to get device CL_DEVICE_VENDOR_ID");
+    error = clGetDeviceInfo(deviceID, CL_DEVICE_VENDOR_ID, sizeof(deviceID_vid),
+                            &deviceID_vid, NULL);
+    test_error(error, "Unable to get device CL_DEVICE_VENDOR_ID");
```

Comment:
It isn't wrong to continue to check the device vendor IDs here, but I also don't think it's needed anymore, since we're checking that the device IDs themselves are equivalent instead.

Reply:
Removed.